test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```
@@ -1052,6 +1052,58 @@ public void testIndexQueriesWithIndexNotReady() throws Throwable
             execute("DROP index " + KEYSPACE + ".testIndex");
         }
     }
+    
+    @Test
+    public void testIndexWritesWithIndexNotReady() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int, ck int, value int, PRIMARY KEY (pk, ck))");
+        String indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + BlockingStubIndex.class.getName() + "'");
+
+        execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+        BlockingStubIndex index = (BlockingStubIndex) getCurrentColumnFamilyStore().indexManager.getIndexByName(indexName);
+        assertEquals(0, index.rowsInserted.size());
+        execute("DROP index " + KEYSPACE + "." + indexName);
+    }
+    
+    @Test // A Bad init could leave an index only accepting reads
+    public void testReadOnlyIndex() throws Throwable
+    {
+        String tableName = createTable("CREATE TABLE %s (pk int, ck int, value int, PRIMARY KEY (pk, ck))");
+        String indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + ReadOnlyIndex.class.getName() + "'");
+        assertTrue(waitForIndex(keyspace(), tableName, indexName));
+
+        execute("SELECT value FROM %s WHERE value = 1");
+        execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+
+        ReadOnlyIndex index = (ReadOnlyIndex) getCurrentColumnFamilyStore().indexManager.getIndexByName(indexName);
+        assertEquals(0, index.rowsInserted.size());
+        execute("DROP index " + KEYSPACE + "." + indexName);
```

We could try a recovery here, and verify that the index starts to accepts writes after that recovery. Also, in `ReadOnlyIndex`, we could throw an exception in `getInitializationTask`, and provide a mock implementation `getRecoveryTaskSupport` that enables writes. Note that forcing an exception during initialization would make us to miss the call to `markIndexBuilt`, so the index won't be added to `SIM#queryableIndexes` and it won't accepts reads independently of its `ReadOnlyIndex#supportsLoad` implementation. That's why I think we need some new logic in `SIM#markIndexFailed` too.